src/python/include/Tangential_complex_interface.h

```diff
@@ -88,8 +88,8 @@ class Tangential_complex_interface {
       num_inconsistencies_ = tangential_complex_->number_of_inconsistent_simplices();
   }
 
-  void create_simplex_tree(Simplex_tree<>* simplex_tree) {
-    tangential_complex_->create_complex<Gudhi::Simplex_tree<Gudhi::Simplex_tree_options_full_featured>>(*simplex_tree);
+  void create_simplex_tree(Simplex_tree<Simplex_tree_options_for_python>* simplex_tree) {
+    tangential_complex_->create_complex<Simplex_tree<Simplex_tree_options_for_python>>(*simplex_tree);
```

Comment:
```suggestion
    tangential_complex_->create_complex<Simplex_tree_interface>(*simplex_tree);
```
Here, when it calls `insert_simplex_and_subfaces`, it calls the method from Simplex_tree_interface and it fails because it is not returning the type required by tangential complex

Reply:
So the issue is that `Simplex_tree_interface::insert_simplex_and_subfaces` overrides the functions of the base class but has an incompatible API? Shouldn't we rename that function so it does not clash with the one in the base class then?

Reply:
Actually, that function does not seem to be used anywhere, can't we remove it?

Reply:
Oh, ok ! That should be some dead code. I removed it and now my suggestion is possible on a6c5e0ae7